### PR TITLE
TST: tox option to exclude scipy

### LINKED ISSUE #15701: CI: test numpy-dev without scipy?
### Description

Currently CI only has a job testing numpy 2.0.0dev with scipy dev installed as well. This skips some tests, only 4 afaics, but 
`test_quantity_ufuncs:test_coverage` is fairly relevant as long as the ufunc namespace keeps getting reshuffled – see e.g. #15699. Should we add a npdev setup without scipy at least until numpy 2.0 is finalised?

### Expected behavior

_No response_

### How to Reproduce

Install dev versions as e.g. described in [#15690](https://github.com/astropy/astropy/issues/15690#issuecomment-1845319381) and then `git uninstall scipy` if it has been installed with `".[test]"`; then run
`pytest astropy/units/tests/test_quantity_non_ufuncs.py -vvk test_coverage`

### Versions

_No response_

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to add tox option to exclude scipy so we can test devdeps without scipy-dev. I opted to run this combo in weekly cron. The setup and cadence are all negotiable.

Also cc @neutrinoceros 

Fixes #15701

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by pllim** [comment]
I guess I cannot force uninstall like this...? Will have to think of another way tomorrow.

**Comment by pllim** [comment]
I have no idea why flynt is failing here. Looks unrelated.

**Comment by pllim** [comment]
OK, looks like I got it: https://github.com/astropy/astropy/actions/runs/7217133913/job/19664457683?pr=15742

```
Package versions: 
Numpy: 2.0.0.dev0+git20231212.9726755
Scipy: not available
Matplotlib: not available
h5py: not available
Pandas: not available
PyERFA: 2.0.1.2.dev9+g0d5ddca
Cython: not available
Scikit-image: not available
pyarrow: not available
asdf-astropy: not available
```

**Review by neutrinoceros — COMMENTED** [review]
It's a great idea, thank you @pllim 
I note that I was expecting this combo to run alongside the "all dev devs" one, not to replace it, but maybe there's a cost to having both that I'm not seeing ?

**Comment by pllim** [comment]
> run alongside the "all dev devs" one

I guess we can. I just thought we already have enough jobs as it is, and we have so much traffic that the devdeps in CI for PR and push basically runs constantly, so no point also calling it again in cron job. But perhaps having both in cron is safer in case this repo gets a slow period.

What do others think?

**Comment by pllim** [comment]
@dhomeier , do you have any opinions here?

**Comment by dhomeier** [comment]
> I guess we can. I just thought we already have enough jobs as it is, and we have so much traffic that the devdeps in CI for PR and push basically runs constantly, so no point also calling it again in cron job. But perhaps having both in cron is safer in case this repo gets a slow period.

I don't have a strong opinion, but also think this can be read both ways – we typically have a dozen or more full runs of ci_workflows every day, so any standard job on top of that in cron_weekly or actually even cron_daily has fairly negligible impact. There is the same job just with py312 indeed, but I'd tend to test those for at least two Python versions (though actually the combination of newest `devdeps` with oldest supported Python may be more interesting.
But I also don't know why e.g. `mpldev` warrants its own extra job in workflows...

Bottom line is probably I'd see cron_weekly as primarily serving the really long and expensive jobs like the exotic archs, while this job could well go into cron_daily – along with the old `py311-test-devdeps` or perhaps in fact a `py310-test-devdeps`.

**Comment by pllim** [comment]
I thought about it but I only want to deal with devdeps failures once a week, so I opted to keep it in weekly but make it a new job with older Python 3.10. Testing cutting age stuff with oldest supported Python seems like we are just asking for trouble.

**Comment by pllim** [comment]
@neutrinoceros and @dhomeier , what do you think now?

https://github.com/astropy/astropy/actions/runs/7465080616/job/20313469760?pr=15742

**Comment by neutrinoceros** [comment]
```
pluggy._manager.PluginValidationError: Plugin 'pytest_mpl' for hook 'pytest_report_header'
hookimpl definition: pytest_report_header(config, startdir)
Argument(s) {'startdir'} are declared in the hookimpl but can not be found in the hookspec
```
If I'm reading this right, this means that `pytest_mpl` isn't compatible with pytest-dev at the moment. At a first glance it doesn't seem like it was reported upstream yet.

**Comment by pllim** [comment]
You can ignore devinfra failures. I have a patch already at https://github.com/matplotlib/pytest-mpl/pull/219 that is being tested at https://github.com/astropy/astropy/pull/15820

**Comment by pllim** [comment]
linkcheck and predeps failures are also unrelated. What is important here is that the new job is successful and did not pull in scipy.

**Comment by neutrinoceros** [comment]
Ah sorry I completely missed the point !
Can we add a step to invoke `pip freeze` right before tests so it's easier to track down in logs ?

**Comment by pllim** [comment]
There is a pip freeze right before the test starts, see https://github.com/astropy/astropy/actions/runs/7465080616/job/20313469760?pr=15742

**Comment by neutrinoceros** [comment]
Please ignore everything I said (apparently) 😅

**Comment by dhomeier** [comment]
> Testing cutting age stuff with oldest supported Python seems like we are just asking for trouble.

That was the point of testing it. ;-)
I was rather thinking of `py311-test-devdeps-noscipy` + `py310-test-devdeps`, but should not matter too much.

And I take your point that it would be enough to deal with this once a week, had only thought about CI resources.

**Comment by pllim** [comment]
@dhomeier , OK I flipped the Python versions as you requested. If it looks good now, can you please approve? Thanks!

**Review by dhomeier — APPROVED** [review]
Thanks, remaining weekly jobs should be no issue!

**Comment by dhomeier** [comment]
How do we enable auto-merge here?

**Comment by pllim** [comment]
Auto merge won't make sense since the new job isn't required. Thanks for the approval. I will keep an eye on it and merge when the optional new job passes.

**Comment by dhomeier** [comment]
Has already, just waiting for the slower archs, so I thought about shortening that...

**Comment by pllim** [comment]
I cancelled the really long jobs and merged.

Hmm... I guess backport isn't critical since auto-backport failed. I'll update the milestone.
